legacy/src/main/java/org/opensearch/sql/legacy/query/planner/physical/node/pointInTime/PointInTime.java

```diff
@@ -0,0 +1,208 @@
+package org.opensearch.sql.legacy.query.planner.physical.node.pointInTime;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Objects;
+import org.opensearch.action.search.SearchResponse;
+import org.opensearch.client.Client;
+import org.opensearch.common.unit.TimeValue;
+import org.opensearch.common.xcontent.XContentType;
+import org.opensearch.core.common.Strings;
+import org.opensearch.index.query.BoolQueryBuilder;
+import org.opensearch.index.query.QueryBuilder;
+import org.opensearch.search.SearchHit;
+import org.opensearch.search.builder.PointInTimeBuilder;
+import org.opensearch.search.sort.FieldSortBuilder;
+import org.opensearch.search.sort.SortOrder;
+import org.opensearch.sql.legacy.domain.Where;
+import org.opensearch.sql.legacy.exception.SqlParseException;
+import org.opensearch.sql.legacy.pit.PointInTimeHandlerImpl;
+import org.opensearch.sql.legacy.query.join.TableInJoinRequestBuilder;
+import org.opensearch.sql.legacy.query.maker.QueryMaker;
+import org.opensearch.sql.legacy.query.planner.core.ExecuteParams;
+import org.opensearch.sql.legacy.query.planner.core.PlanNode;
+import org.opensearch.sql.legacy.query.planner.physical.Row;
+import org.opensearch.sql.legacy.query.planner.physical.estimation.Cost;
+import org.opensearch.sql.legacy.query.planner.physical.node.BatchPhysicalOperator;
+import org.opensearch.sql.legacy.query.planner.physical.node.SearchHitRow;
+import org.opensearch.sql.legacy.query.planner.resource.ResourceManager;
+
+/** OpenSearch Search API with Point in time as physical implementation of TableScan */
+public class PointInTime extends BatchPhysicalOperator<SearchHit> {
```

Comment:
I see many similar logic as `Scroll` operator. Can we add a base class Paginate?

For example, for prefetch, Scroll and PointInTime only needs to implement loadFirstBatch and loadNextBatch?

Reply:
Added Paginate.java base class